server/src/main/java/keywhiz/service/permissions/AllowClientSecretCreationPermissionCheck.java

```diff
@@ -0,0 +1,18 @@
+package keywhiz.service.permissions;
+
+import keywhiz.api.model.Client;
+import keywhiz.api.model.Secret;
+
+public class AllowClientSecretCreationPermissionCheck implements PermissionCheck {
+  @Override
+  public boolean isAllowed(Object source, String action, Object target) {
+    return false;
+  }
+
+  @Override
+  public boolean isAllowedForTargetType(Object source, String action, Class<?> targetType) {
+    return
+        Client.class.isAssignableFrom(source.getClass()) &&
+        Secret.class.isAssignableFrom(targetType);
+  }
```

Comment:
Technically the first line is identical to `source instanceof Client`, but I liked the symmetry with the second line where `isAssignableFrom()` is required because the class is a variable (instanceof is compile-time, isAssignableFrom is run-time).